tests/rust/src/bin/stdio.rs

```diff
@@ -1,13 +1,47 @@
-use wasi_tests::{STDERR_FD, STDIN_FD, STDOUT_FD};
+use std::{env, process};
 
-unsafe fn test_stdio() {
-    for fd in &[STDIN_FD, STDOUT_FD, STDERR_FD] {
-        wasi::fd_fdstat_get(*fd).expect("fd_fdstat_get on stdio");
-        wasi::fd_renumber(*fd, *fd + 100).expect("renumbering stdio");
+use wasi_tests::{create_tmp_dir, open_scratch_directory, STDERR_FD, STDIN_FD, STDOUT_FD};
+
+unsafe fn test_stdio(dir_fd: wasi::Fd) {
+    for stdio_from_fd in &[STDIN_FD, STDOUT_FD, STDERR_FD] {
+        let stdio_to_fd =
+            wasi::path_open(dir_fd, 0, "file", wasi::OFLAGS_CREAT, 0, 0, 0).expect("open file");
```

Comment:
Any temporary files should have a `.cleanup` suffix. See https://github.com/WebAssembly/wasi-testsuite/blob/main/doc/specification.md

Reply:
Done